docs/design/proposals/healer-controller.md

```diff
@@ -0,0 +1,120 @@
+# Introduction
+There is [external health monitoring](https://github.com/kubernetes-csi/external-health-monitor/) which does mostly similar stuff, but actions are not in the scope of it, all it can do is log the health condition of the mounts.
+
+# Volume Healer Controller
+Before we begin it is noteworthy to know that the existing healer logic remains the same.
+
+**Now the obvious question we expect is "what do we achieve with the new controller?"**
+* Moving the k8s specific code out of ceph-csi, to make ceph-csi orchestrator independent.
+* Achieve a way to monitor the health of NBD volumes.
+* We currently do not have a way to act on failed/crashed NBD volumes with the existing healer logic. Once the controller detects that the volume is not healthy, it will have an opportunity to act on it.
+* This will be a generic framework for any userspace mounters added in the future, example ceph-fuse
```

Comment:
csi-addons is the framework! Phrase it differently, explain what the framework offers, and only after that describe an example with Ceph-CSI and rbd-nbb (and maybe mention that the healing is optional, e.g. krbd would not need it?)